temporal-sdk/src/main/java/io/temporal/common/converter/AbstractProtobufPayloadConverter.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.common.converter;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.MessageOrBuilder;
+import io.temporal.api.common.v1.Payload;
+import java.lang.reflect.Method;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractProtobufPayloadConverter {
```

Comment:
I'd name this `ProtobufPayloadConverter` and rename `ProtobufPayloadConverter -> ProtobufBinaryPayloadConverter` but that would be a breaking change.

Reply:
Honestly, I think that `ProtobufPayloadConverter` is a better name than `ProtobufBinaryPayloadConverter `. Because for me protobuf is a binary format and this doesn't really need Binary here. Also, it will make it clear that this is "the main one" for people who want proto serialization. ProtoJson is some strange thing, that hopefully will not be used much by anybody, because it's a mix of the worst things of both worlds at the end.

Reply:
Haha, makes sense 😄